app/subscriber/src/components/user-profile/UserProfile.tsx

```diff
@@ -13,30 +16,71 @@ import * as styled from './styled';
 export const UserProfile: React.FC = () => {
   const keycloak = useKeycloakWrapper();
   const { width } = useWindowSize();
+  const [{ profile, impersonate }, { storeImpersonate }] = useProfileStore();
+  const { getUser } = useUsers();
+
+  const [profileMenu, setProfileMenu] = React.useState(false);
+
+  const isAdmin = keycloak.hasClaim(Claim.administrator);
+
+  React.useEffect(() => {
+    if (
+      profile?.preferences?.impersonate &&
+      profile?.preferences?.impersonate !== impersonate?.key
+    ) {
+      getUser()
+        .then((user) => {
+          storeImpersonate(user);
+        })
+        .catch(() => {});
+    } else if (!profile?.preferences?.impersonate) {
+      storeImpersonate(undefined);
+    }
+    // getUser will cause a double request.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [profile?.preferences?.impersonate, impersonate, storeImpersonate]);
 
   return (
     <styled.UserProfile>
-      <Row data-tooltip-id="my-info" direction="row" className="username-info">
-        <Show visible={!!width && width > ScreenSizes.Mobile}>
-          <FaChevronCircleDown size={15} />
-        </Show>
-        <div>{keycloak.getDisplayName()}</div>
+      <Row
+        data-tooltip-id="my-info"
+        direction="row"
+        className={`username-info${impersonate ? ' impersonate' : ''}`}
+      >
+        <div>
+          <Show visible={!!width && width > ScreenSizes.Mobile}>
+            <FaChevronCircleDown size={15} />
+          </Show>
+          <div>{impersonate ? impersonate.displayName : keycloak.getDisplayName()}</div>
+        </div>
       </Row>
       <Tooltip
         clickable
         variant="light"
         className="folder-menu"
         place="bottom"
         openOnClick
+        closeOnEsc
+        isOpen={profileMenu}
+        setIsOpen={setProfileMenu}
         style={{ opacity: '1', boxShadow: '0 0 8px #464545', zIndex: '999' }}
         id="my-info"
       >
         <h1>My Profile</h1>
         <hr />
         <ul>
           <li>
-            <a href="/colleagues">Colleagues</a>
+            <Link to="/colleagues" onClick={(e) => setProfileMenu(false)}>
```

Comment:
Don't use anchor tags as they refresh the page.  Use `<Link/>` because we're using an SPA.